department-of-veterans-affairs/abd-vro
# Mcp 1938 - Add MAS API calls for collection status and order exam.

Add MAS API call service to call collection status and order exam.

## What was the problem?
For MAS integration, we need to call MAS APIs.

Associated tickets or Slack threads:

MCP-1938 and MCP-1945

## How does this fix it?[^1]

Add MAS API call service in service:provider for now. It may need to be moved when we have more information on MAS API.

## How to test this PR
- Step 1 Build application, go swagger_ui page, try MAS Claim Request (/v1/notifyVROAutomatedClaimDetails).
- Step 2[^secrel] Check app log, you should see something like the following.
   [cation-exchange] g.v.v.s.provider.MasPollingProcessor     : Checking collection status for collection 999.
   INFO 1 --- [cation-exchange] g.v.v.s.p.mas.service.MasApiService      : Call http://localhost:5000/pca/api/dev/pcCheckCollectionStatus?collectionIds=999 to get MAS collection status.
2022-10-13 22:15:39.889 ERROR 1 --- [cation-exchange] g.v.v.s.p.mas.service.MasApiService      : Failed to get collection status.

Note: the call to MAS here will be failed, since we have no MAS API service/server. As we don't have the MAS connection information, I put localhost:5000 here. We'll need a MAS API server.



[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
